bokeh/_json_encoder.py

```diff
@@ -79,4 +79,4 @@ def default(self, obj):
             return self.transform_python_types(obj)
 
 def serialize_json(obj, encoder=BokehJSONEncoder, **kwargs):
-    return json.dumps(obj, cls=encoder, allow_nan=False, **kwargs)
+    return json.dumps(obj, cls=encoder, allow_nan=True, **kwargs)
```

Comment:
Encapsulating the data somehow. JavaScript supports the the `NaN` and `Infinity` values, we just need a way to pass it through JSON. @bryevdv has mentioned using typed arrays, etc. I've only recently learned about them, but I am anxious to investigate.